platform-hub-api/app/models/kubernetes_token.rb

```diff
@@ -69,23 +71,25 @@ def obfuscated_token
     val[0..30].gsub(/\w/, 'X') + val[31..35]
   end
 
-  def owner
-    if self.user?
-      case self.tokenable
-      when Identity
-        self.tokenable.user
-      end
-    end
-  end
-
   def privileged?
     expire_privileged_at.present?
   end
 
-  def escalate(privileged_group, expire_in_secs = 600)
-    update_attributes(
+  def escalate(privileged_group_name, expire_in_secs = 600)
+    group = KubernetesGroup.where(name: privileged_group_name).first
+
+    if group.blank?
+      raise ActiveRecord::RecordNotFound
+    end
+
+    unless is_group_valid?(group, allow_privileged_groups: true)
+      raise ArgumentError, "specified privileged group '#{privileged_group_name}' is not a valid group for this token (it may not be allocated yet)"
+    end
+
+    # Update the db directly, bypassing validations
```

Comment:
Why do we want to skip validations here?

Reply:
The issue here is the `allowed_groups_only` validation check – currently, it won't allow privileged groups to be set (to prevent project admins from explicitly setting this when creating/editing tokens from the UI/API). So this method fails if we run this validation (because it's trying to set a privileged group).

Reply:
I see. Maybe worth putting some extra context in the comment describing why we chose to do that.

Reply:
Good idea.